src/main/java/org/mockito/internal/creation/bytebuddy/SubclassBytecodeGenerator.java

```diff
@@ -90,15 +90,53 @@ protected SubclassBytecodeGenerator(
         handler = ModuleHandler.make(byteBuddy, loader, random);
     }
 
+    private static boolean needsSamePackageClassLoader(MockFeatures<?> features) {
+        if (!Modifier.isPublic(features.mockedType.getModifiers())
+                || !features.mockedType.isInterface()) {
+            // The mocked type is package private or is not an interface and thus may contain
+            // package
+            // private methods.
+            return true;
+        }
+        for (Class<?> iface : features.interfaces) {
```

Comment:
Interfaces might declare interfaces of their own.

Reply:
That's OK, as all interface methods are public, and a public interface that extends a package-private interface doesn't require the implementation to be in the same package. All that matters for the generated type is that the directly implemented interfaces are accessible.

Reply:
Of course, my bad. I was thinking that you iterated over the interfaces of the class but it's of course the directly implemented ones that matter.